packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -1303,27 +1311,44 @@ class SlideshowView extends GalleryComponent {
   //-----------------------------------------| REACT |--------------------------------------------//
 
   blockAutoSlideshowIfNeeded(props = this.props) {
-    const { isGalleryInHover } = props;
-    const { pauseAutoSlideshowClicked, shouldBlockAutoSlideshow, isInView } =
+    const { isGalleryInHover, styleParams } = props;
+    const { pauseAutoSlideshowClicked, shouldBlockAutoSlideshow, isInView,isInFocus } =
       this.state;
     let should = false;
     if (!isInView || pauseAutoSlideshowClicked) {
       should = true;
     } else if (
-      isGalleryInHover &&
-      props.styleParams.pauseAutoSlideshowOnHover
+      styleParams.pauseAutoSlideshowOnHover
     ) {
-      should = true;
+      should = isGalleryInHover || isInFocus && styleParams.isAccessible;
```

Comment:
Whats different between SP.isAccessible and SP.pauseAutoSlide...Hover ? (or the others?)
Why is one in an if and the rest in the assignment logic?
I would rather have them all in the if like they were before. if the logic is more specific now it should be there.
if the || is bothering you, this can be 2 else if
first one for the hover logic, and 2nd for the focus logic.
both assigning true to should.

Reply:
SP.pauseAutoSlideOnHover is the condition that should appear in both cases, and SP.isAccessible should only appear for the isInFocus state, It will also work the way you mentioned here, it's just my writing preference but I'll change it.